### PR TITLE
Feathr Web UI MVP Prototype

### LINKED ISSUE #215: Feathr Web UI MVP
Need a web ui as enhancement to cli for feature registry, search, lineage features, etc. 

Following should be included for MVP prototype:
- [ ] Setup UX framework: site global style, layout, menu, header, etc.
- [ ] Account login
- [ ] Data sources listing
- [ ] Feature listing and search
- [ ] Feature lineage flow chart
- [ ] Feature/Project assignment

### PR DESCRIPTION
**Pull request by blrchen**
This PR initialize the fundamental codebase for feathr web UI development, to support following up UX user stories implementation. The codebase is initialized with [Create React App](https://github.com/facebook/create-react-app), plus following code implementation: 

### UX framework related implemented

- [x] React recommended eslint settings are enabled.
- [x] Add a workflow for azure container app  CI/CD
- [x] Add React components for global static contents: site layout, menu sidebar, header, footer, etc. 

### User stories implemented

- [x] Azure AD Account login
- [x] Data sources listing
- [x] List features by project
- [x] View feature metadata properties
- [x] View Feature lineage flow chart

### Docs

- [x] A README.md is added under ui directory for general ui dev get start doc

### Built on

| project | license |
|-|-|
|@azure/msal-browser,@azure/msal-react |MIT|
|antd |MIT|
|axios |MIT|
|dagre |MIT|
|react,react-dom,react-router|MIT|
|react-flow-renderer|MIT|
|react-query|MIT|
|react-resizable|MIT|

Closes #215

### CODE REVIEW

File: ui/src/api/api.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import Axios from "axios";
+import { Features, IDataSource, IFeature, IFeatureDetail, IFeatureLineage } from "../models/model";
+// import lineage from "./mock-lineage.json";
```
Comment (jaymo001) [inline ui/src/api/api.tsx:3]: delete

File: ui/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Build and run locally
+
+Make sure you have the latest node LTS installed
+
+Run the following commands in your terminal
+
+```bash
+cd ui
+npm install
```
Comment (jaymo001) [inline ui/README.md:13]: I'm seeing this error when running npm start:
Compiled with problems:X

ERROR

Must use import to load ES Module:
...

Could you add a note here saying: 
node -v should display a version > 16.0 ? (12.x does not work for me).

File: ui/public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>Azure Feature Store</title>
```
Comment (jaymo001) [inline ui/public/index.html:27]: Feathr?

File: ui/src/components/header/header.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  const onClickLogout = () => {
+    instance.logoutRedirect().catch(e => {
+      console.error(e);
+    });
+  }
+
+  const { Paragraph } = Typography;
+  return (
+    <PageHeader
+      title={ `Hello ${ name }, welcome to Azure Feature Store` }
```
Comment (jaymo001) [inline ui/src/components/header/header.tsx:28]: Feathr?

File: ui/public/index.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <!--
+      This HTML file is a template.
```
Comment (jaymo001) [inline ui/public/index.html:33]: Delete

File: ui/public/index.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+    <!--
```
Comment (jaymo001) [inline ui/public/index.html:13]: Delete

File: ui/public/index.html [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      manifest.json provides metadata used when your web app is installed on a
+      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
+    -->
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <!--
```
Comment (jaymo001) [inline ui/public/index.html:18]: Delete

File: ui/src/react-app-env.d.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (jaymo001) [inline ui/src/react-app-env.d.ts:1]: Delete

File: ui/src/components/dataSourceList.tsx [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    },
+    {
+      title: <div style={ { userSelect: "none" } }>Consumers</div>,
+      dataIndex: 'name',
+      key: 'name',
+      render: () => {
+        return (
+          <Button type={ "link" } onClick={ () => {
+            navigateTo(`/features`)
+          } }>9</Button>
```
Comment (jaymo001) [inline ui/src/components/dataSourceList.tsx:87]: Hard-coded?

**Comment by hangfei** [comment]
For foundational PRs, a dev guide is required.

File: .github/workflows/azure-static-web-apps-thankful-mud-0addb6c00.yml [hunk 10]
```diff
@@ -0,0 +1,47 @@
+name: Azure Static Web Apps CI/CD
```
Comment (hangfei) [inline .github/workflows/azure-static-web-apps-thankful-mud-0addb6c00.yml:1]: is the file name meant to be random?

File: ui/.eslintrc [hunk 11]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (hangfei) [inline ui/.eslintrc:1]: documenting what's this for.

File: ui/package.json [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    "antd": "^4.3.0",
+    "axios": "^0.27.2",
+    "dagre": "^0.8.5",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-flow-renderer": "^10.2.0",
+    "react-query": "^3.2.0",
+    "react-resizable": "^3.0.4",
+    "react-router": "^5.0.1",
+    "react-router-dom": "^5.1.2"
```
Comment (hangfei) [inline ui/package.json:17]: license of these dependencies should be in the description.

**Review by hangfei — COMMENTED** [review]
Code documentation is needed for future devs.  See https://github.com/linkedin/feathr/blob/main/feathr_project/feathr/client.py as an example.

File: .github/workflows/azure-static-web-apps-thankful-mud-0addb6c00.yml (hunk 10 above)
Comment (blrchen) [reply .github/workflows/azure-static-web-apps-thankful-mud-0addb6c00.yml]: The random string refers to the container instances it refers to, it has been removed. The secret name in yml has been renamed to be more meaningful as well.

File: ui/public/index.html (hunk 6 above)
Comment (blrchen) [reply ui/public/index.html:13]: Deleted

File: ui/public/index.html (hunk 7 above)
Comment (blrchen) [reply ui/public/index.html:18]: Deleted

File: ui/public/index.html (hunk 3 above)
Comment (blrchen) [reply ui/public/index.html:27]: Changed Azure Feature Store to Feathr Feature Store

File: ui/src/api/api.tsx (hunk 1 above)
Comment (blrchen) [reply ui/src/api/api.tsx:3]: Deleted

File: ui/src/components/dataSourceList.tsx (hunk 9 above)
Comment (blrchen) [reply ui/src/components/dataSourceList.tsx:87]: The consumer # was there for demo purpose. This has been removed and can be add later with backend implementation

File: ui/src/components/header/header.tsx (hunk 4 above)
Comment (blrchen) [reply ui/src/components/header/header.tsx:28]: Changed Azure Feature Store to Feathr Feature Store

File: ui/src/react-app-env.d.ts (hunk 8 above)
Comment (blrchen) [reply ui/src/react-app-env.d.ts:1]: It's better not remove this file per https://stackoverflow.com/questions/67262914/what-is-the-react-app-env-d-ts-in-a-react-typescript-project-for

File: ui/package.json (hunk 12 above)
Comment (blrchen) [reply ui/package.json:17]: Add a section in pr description to cover license. Basically, all down streaming dependencies are MIT licensed.

File: ui/.eslintrc (hunk 11 above)
Comment (blrchen) [reply ui/.eslintrc:1]: Add a linting section in README.md under ui directory to cover this.

**Review by windoze — DISMISSED** [review]

**Comment by hangfei** [comment]
Thanks for adding the license description.
In the license section, what's antd?

**Review by jaymo001 — DISMISSED** [review]

**Review by xiaoyongzhu — APPROVED** [review]

**Pull request merged**